warden/src/assignment_computation.rs

```diff
@@ -1,30 +1,204 @@
-use common::{host_info::HostInfo, keyspace_id::KeyspaceId};
+use std::collections::{BinaryHeap, HashMap};
+use std::sync::Arc;
+use std::{collections::HashSet, ops::Deref, sync::Mutex};
+
+use common::host_info::HostInfo;
 use proto::warden::WardenUpdate;
+use std::cmp::{Ordering, Reverse};
+use std::hash::{Hash, Hasher};
 use tokio::sync::broadcast::Receiver;
+use tokio_util::sync::CancellationToken;
 use tonic::Status;
+use tracing::{debug, info};
 
 use crate::persistence::{RangeAssignment, RangeInfo};
 
+// TODO(purujit): Convert this to a configuration.
+const MIN_NUM_RANGE_SERVERS: usize = 1;
+
+#[derive(Clone, Debug)]
+struct HostInfoWrapper(HostInfo);
```

Comment:
Could you add a comment explaining the purpose of this wrapper?

Reply:
Done.